sql/src/main/codegen/includes/replace.ftl

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Taken from syntax of SqlInsert statement from calcite parser, edited for replace syntax
+SqlNode DruidSqlReplaceEof() :
+{
+    SqlNode table;
+    SqlNode source;
+    SqlNodeList columnList = null;
+    final Span s;
+    SqlInsert sqlInsert;
+    // Using fully qualified name for Pair class, since Calcite also has a same class name being used in the Parser.jj
+    org.apache.druid.java.util.common.Pair<Granularity, String> partitionedBy = new org.apache.druid.java.util.common.Pair(null, null);
+    final Pair<SqlNodeList, SqlNodeList> p;
+    final SqlNode replaceTimeQuery;
+}
+{
+    <REPLACE> { s = span(); }
+    table = CompoundIdentifier()
+    [
+        p = ParenthesizedCompoundIdentifierList() {
+            if (p.left.size() > 0) {
+                columnList = p.left;
+            }
+        }
+    ]
+    <DELETE>
+    (
+      <ALL> { replaceTimeQuery = SqlLiteral.createCharString("ALL", getPos()); }
```

Comment:
`<ALL> | WhereOpt()` can be made a rule like `DeleteOpts()` or `PartitionSpec()`. Would make adding more formats easier.

Reply:
Done